src/main/java/uk/ac/cam/cl/dtg/isaac/api/EventsFacade.java

```diff
@@ -486,6 +497,50 @@ public final Response getEventBookingByEventId(@Context final HttpServletRequest
         }
     }
 
+    /** gets a list of event bookings based on a given group id.
+     *
+     */
+    @GET
+    @Path("{event_id}/bookings/for_group/{group_id}")
+    @Produces(MediaType.APPLICATION_JSON)
+    @GZIP
+    @ApiOperation(value = "List event bookings for a specific event and group.")
+    public final Response getEventBookingForGivenGroup(@Context final HttpServletRequest request,
+                                                       @PathParam("event_id") final String eventId,
+                                                       @PathParam("group_id") final String groupId) {
+        try {
+            RegisteredUserDTO currentUser = userManager.getCurrentRegisteredUser(request);
+            if (!Arrays.asList(Role.TEACHER, Role.EVENT_LEADER, Role.EVENT_MANAGER, Role.ADMIN).contains(currentUser.getRole())) {
+                return SegueErrorResponse.getIncorrectRoleResponse();
+            }
+
+            UserGroupDTO group = groupManager.getGroupById(Long.parseLong(groupId));
+            // af599 TODO: Might make sense to check whether the currentUser is allowed to interact with the group
+
+            List<Long> groupMemberIds = groupManager.getUsersInGroup(group)
+                    .stream().map(RegisteredUserDTO::getId)
+                    .collect(Collectors.toList());
+
+            // Filter eventBookings based on whether the booked user is a member of the given group
+            List<EventBookingDTO> eventBookings = bookingManager.getBookingByEventId(eventId)
+                    .stream().filter(booking -> groupMemberIds.contains(booking.getUserBooked().getId()))
+                    .collect(Collectors.toList());
+
+            // Event leaders are only allowed to see the bookings of connected users
+            if (Role.EVENT_LEADER.equals(currentUser.getRole())) {
```

Comment:
Should this check not apply to teachers too? I.e. teachers should only see information about the users they have been given permission to view?